docs/conventions/component.md

```diff
@@ -0,0 +1,527 @@
+
+Apart from [rules that should be applied to all Twig files](twig), there are some specifications for components:
+
+### Generic component
+
+Generic component Twig file defines all the markup for the component. It should contain all the logic required to use the component, as it is intended to be used by systems components (or anything based on Twig).
+
+This Twig file consists of different sections (see link above for more details):
+
+* Parameters/Blocks: list of exposed parameters and blocks
+* Internal properties: definition of custom variables
+* Internal logic: preprocess and other custom logic
+* Print: HTML markup
+
+Example:
+_(File name: generic-component-blockquote.twig)_
+
+```twig
+{#
+  Parameters:
+    - "body" (string) (default: '')
+    - "author" (string) (default: '')
+    - "extra_classes" (string) (default: '')
+    - "extra_attributes" (array) (default: ''): format: [{ 'name': 'name_of_the_attribute', 'value': 'value_of_the_attribute'}])
+#}
+
+{# Internal properties #}
+
+{% set _body = body|default('') %}
+{% set _author = author|default('') %}
+{% set _css_class = 'ecl-blockquote' %}
+{% set _extra_attributes = '' %}
+
+{# Internal logic - Process properties #}
+
+{% if extra_classes is defined %}
+  {% set _css_class = _css_class ~ ' ' ~ extra_classes %}
+{% endif %}
+
+{% if extra_attributes is defined %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~'"' %}
+  {% endfor %}
+{% endif %}
+
+{# Print the result  #}
+
+<blockquote class="{\{ _css_class }}"{\{ _extra_attributes|raw }}>
+  <p class="ecl-paragraph ecl-blockquote__body">{{ _body }}</p>
+  {% if _author is not empty %}
+  <footer class="ecl-blockquote__author">―&thinsp;<cite>{{ _author }}</cite></footer>
+  {% endif %}
+</blockquote>
+```
+
+### Systems component
+
+There are 2 possibilities for system-based components: they use the exact same markup as their corresponding generic/base component, or they alter it.
+
+In both cases, **Twig file should always start with available parameters**.
+
+#### Same markup
+
+This is the easy case: system component just has to include generic Twig file.
+
+Example:
+_(File name: ec-component-blockquote.twig)_
+
+```twig
+{#
+  Parameters:
+    - "body" (string) (default: '')
+    - "author" (string) (default: '')
+    - "extra_classes" (string) (default: '')
+    - "extra_attributes" (array) (default: ''): format: [{ 'name': 'name_of_the_attribute', 'value': 'value_of_the_attribute'}])
+#}
+
+{% include '@ecl/generic-component-blockquote' %}
+```
+
+#### Different markup
+
+As soon as there is a difference in markup (even a small one), specific component has to provide the whole markup (instead of using generic one).
+This is the same structure as generic Twig file.
```

Comment:
`This is the same structure as generic Twig file.` -> `The structure is the same as for the generic Twig file.`